internal/topo/planner/planner.go

```diff
@@ -266,7 +266,7 @@ func createLogicalPlan(stmt *ast.SelectStatement, opt *api.RuleOption, store kv.
 			p.SetChildren(append(children, tableChildren...))
 			children = []LogicalPlan{p}
 		} else if w == nil {
-			return nil, errors.New("need to run stream join in windows")
+			return nil, errors.New("need to run stream join within a window by using the GROUP BY WINDOW clause")
```

Comment:
need to run JOIN statement within a window by using the GROUP BY WINDOW clause

Reply:
yes. otherwise, the rule will receive one message from a single stream which has no meaning of join. A window is needed to hold multiple messages so that they could be from multiple streams to join.